tests/network/http/websocket_test.cpp

```diff
@@ -39,7 +39,7 @@ BOOST_AUTO_TEST_CASE(websocket_test)
         try {
             c_conn->send_message( "again" );
             BOOST_FAIL("expected assertion failure");
-        } catch (const fc::assert_exception& e) {
+        } catch (const fc::exception& e) {
```

Comment:
Why this?

Reply:
We've just fixed a failing test. It seems that at some point the underlying logic has changed to no longer throw a fc::assert_exception and the test went dead. 

Please note the pattern: the code is not using BOOST_EXPECT_THROW as it could have thus harming readability.

Reply:
Makes sense. Bug fixes are OK.
However, you named this PR as fixes to "sanitizer issues" so I mostly checked if your code has changed logic.

Reply:
Yeah, we lumped in lots of warning fixes as well... We are mostly using GCC7 and it's being very loud about every minute issue it thinks it has with the code.